extension/intents/forms/formsContentScript.js

```diff
@@ -192,6 +192,28 @@ this.dictationContentScript = (function() {
     return focusDirection(-1);
   });
 
+  communicate.register("turnSelectionIntoLink", async message => {
+    const url = message.url;
+    const text = message.text;
+    let el;
+    if (!isPasteable(document.activeElement)) {
+      for (const selector of PASTE_SELECTORS) {
+        el = document.querySelector(selector);
+        if (el && isInViewport(el)) {
+          break;
+        }
+      }
+    } else {
+      el = document.activeElement;
+    }
+    if (el.hasAttribute("contenteditable")) {
```

Comment:
Oh, I see the issue here. So with contenteditable you might have something like:

```html
<div contenteditable="1">
  <p>
    Here is some text I can edit.
  </p>
</div>
```

And the selection will be inside the `<p>`. So you have to look at the parent, like:

```js
let isContentEditable = false;
let checkElement = el;
while (checkElement) {
  if (checkElement.hasAttribute("contenteditable")) {
    isContentEditable = true;
    break;
  }
  checkElement = checkElement.parentNode;
}
```